zio-http/src/main/scala/zhttp/http/HeadersHelpers.scala

```diff
@@ -26,6 +31,16 @@ private[zhttp] trait HeadersHelpers { self: HasHeaders =>
     }
   }
 
+  def getCharSet: Option[Charset] = {
+    val charSetList: List[Charset] = headers
+      .filter(_.value.toString.contains("charset="))
+      .map(x => JHttpUtil.getCharset(x.value, HTTP_CHARSET))
+    charSetList match {
+      case ::(head, _) => Some(head)
+      case Nil         => None
+    }
+  }
```

Comment:
This code can be simplified a bit —

I think we can pass the complete request directly to `HttpUtil.getCharset` and since you are also providing a default, it doesn't need to return an option.